File: DomainObjects/Player.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;
using static DomainObjects.StatTypes;

namespace DomainObjects
{
    public class Player : Person
    {
        public int Id { get; set; } // Primary key for EF Core

        public int? TeamId { get; set; } // Foreign key to Team (nullable - player might not be on a team)

        public Positions Position { get; set; }

        public int Number { get; set; }

        public string? Height { get; set; }

        public int Weight { get; set; }

        public int Age { get; set; }

        public int Exp { get; set; }

        public string? College { get; set; }

        // JSON backing fields
        public string StatsJson { get; set; } = "{}";
        public string SeasonStatsJson { get; set; } = "{}";
        public string CareerStatsJson { get; set; } = "{}";

        [NotMapped]
        public Dictionary<PlayerStatType, int> Stats 
        { 
            get => DeserializeStats(StatsJson);
            set => StatsJson = SerializeStats(value);
        }

        [NotMapped]
        public Dictionary<PlayerStatType, int> SeasonStats 
        { 
            get => DeserializeStats(SeasonStatsJson);
            set => SeasonStatsJson = SerializeStats(value);
        }

        // Realistic attributes
        public int Speed { get; set; } // 0-100

        public int Strength { get; set; } // 0-100

        public int Agility { get; set; } // 0-100

        public int Awareness { get; set; } // 0-100

        public int Fragility { get; set; } = 50; // 0-100 (higher = more injury-prone), defaults to 50

        public int Morale { get; set; } // 0-100

        public int Discipline { get; set; } // 0-100 (higher = fewer penalties)

        // Position-specific skills
        public int Passing { get; set; } // QB

        public int Catching { get; set; } // WR, TE, RB

        public int Rushing { get; set; } // RB, QB

        public int Blocking { get; set; } // OL, TE, FB

        public int Tackling { get; set; } // DL, LB, S, CB

        public int Coverage { get; set; } // CB, S, LB

        public int Kicking { get; set; } // K, P

        // Career stats
        [NotMapped]
        public Dictionary<PlayerStatType, int> CareerStats 
        { 
            get => DeserializeStats(CareerStatsJson);
            set => CareerStatsJson = SerializeStats(value);
        }

        public bool IsRetired { get; set; }

        public int ContractYears { get; set; }

        public int Salary { get; set; }

        public int Potential { get; set; } // 0-100

        public int Progression { get; set; } // 0-100

        public int Health { get; set; } // 0-100

        // Injury tracking

        /// <summary>
        /// Gets or sets current active injury for this player (null if not injured).
        /// </summary>
        public Injury? CurrentInjury { get; set; }

        /// <summary>
        /// Gets a value indicating whether whether the player is currently injured and unavailable.
        /// </summary>
        public bool IsInjured => CurrentInjury != null;

        private Dictionary<PlayerStatType, int> DeserializeStats(string json)
        {
            if (string.IsNullOrEmpty(json)) return new Dictionary<PlayerStatType, int>();
            try 
            {
                var stringDict = JsonSerializer.Deserialize<Dictionary<string, int>>(json);
                var result = new Dictionary<PlayerStatType, int>();
                if (stringDict != null)
                {
                    foreach (var kvp in stringDict)
                    {
                        if (Enum.TryParse<PlayerStatType>(kvp.Key, out var statType))
                        {
                            result[statType] = kvp.Value;
                        }
                    }
                }
                return result;
            }
            catch { return new Dictionary<PlayerStatType, int>(); }
        }

        private string SerializeStats(Dictionary<PlayerStatType, int> stats)
        {
            if (stats == null) return "{}";
            var stringDict = new Dictionary<string, int>();
            foreach (var kvp in stats)
            {
                stringDict[kvp.Key.ToString()] = kvp.Value;
            }
            return JsonSerializer.Serialize(stringDict);
        }
    }
}

File: gridiron-web/src/types/Player.ts
import { Position } from './enums';

export interface Player {
  id: number;
  teamId: number | null;
  position: Position;
  number: number;
  firstName: string;
  lastName: string;
  age: number;
  exp: number; // years of experience
  height: string;
  college: string;

  // General attributes (0-100)
  speed: number;
  strength: number;
  agility: number;
  awareness: number;
  fragility: number;
  morale: number;
  discipline: number;

  // Position-specific skills (0-100)
  passing: number;    // QB
  catching: number;   // WR/TE/RB
  rushing: number;    // RB/QB
  blocking: number;   // OL/TE
  tackling: number;   // Defense
  coverage: number;   // DB/LB
  kicking: number;    // K/P

  // Contract
  contractYears: number;
  salary: number;

  // Development
  potential: number;
  progression: number;

  // Status
  health: number;
  isRetired: boolean;
  isInjured: boolean;
}

export interface PlayerStats {
  // Passing
  passAttempts?: number;
  passCompletions?: number;
  passYards?: number;
  passTouchdowns?: number;
  interceptions?: number;
  sacks?: number;

  // Rushing
  rushAttempts?: number;
  rushYards?: number;
  rushTouchdowns?: number;
  fumbles?: number;

  // Receiving
  receptions?: number;
  receivingYards?: number;
  receivingTouchdowns?: number;

  // Defense
  tackles?: number;
  tacklesForLoss?: number;
  defensiveInterceptions?: number;

  // Kicking
  fieldGoalAttempts?: number;
  fieldGoalsMade?: number;
  extraPointAttempts?: number;
  extraPointsMade?: number;
}
